packages/bot-skeleton/src/scratch/blocks/Binary/Trade Definition/trade_definition_multiplier.js

```diff
@@ -0,0 +1,224 @@
+import DBotStore from '../../../dbot-store';
+import { runGroupedEvents, runIrreversibleEvents } from '../../../utils';
+import { config } from '../../../../constants/config';
+import ApiHelpers from '../../../../services/api/api-helpers';
+
+Blockly.Blocks.trade_definition_multiplier = {
+    init() {
+        this.jsonInit(this.definition());
+    },
+    definition() {
+        return {
+            message0: localize('Multiplier: {{ multiplier }}', {
+                multiplier: '%1',
+            }),
+            message1: `${localize('Stake')}: %1 %2`,
+            message2: '%1',
+            message3: '%1',
+            args0: [
+                {
+                    type: 'field_dropdown',
+                    name: 'MULTIPLIERTYPE_LIST',
+                    options: [['', '']],
+                },
+            ],
+            args1: [
+                {
+                    type: 'field_label',
+                    name: 'CURRENCY_LIST',
+                    text: getCurrencyDisplayCode(config.lists.CURRENCY[0]),
+                },
+                {
+                    type: 'input_value',
+                    name: 'AMOUNT',
+                    check: 'Number',
+                },
+            ],
+            args2: [
+                {
+                    type: 'input_dummy',
+                },
+            ],
+            args3: [
+                {
+                    type: 'input_statement',
+                    name: 'MULTIPLIER_PARAMS',
+                },
+            ],
+            colour: Blockly.Colours.Special1.colour,
+            colourSecondary: Blockly.Colours.Special1.colourSecondary,
+            colourTertiary: Blockly.Colours.Special1.colourTertiary,
+            previousStatement: null,
+            nextStatement: null,
+            tooltip: localize('Define your trade options such as mulitpliers and stake.'),
+            category: Blockly.Categories.Trade_Definition,
+        };
+    },
+    meta() {
+        return {
+            display_name: localize('Multiplier trade options'),
+            description: localize('Define your trade options such as mulitplier and stake.'),
+        };
+    },
+    onchange(event) {
+        if (!this.workspace || this.isInFlyout || this.workspace.isDragging()) {
+            return;
+        }
+        if (
+            (event.type === Blockly.Events.BLOCK_CREATE && event.ids.includes(this.id)) ||
+            event.type === Blockly.Events.END_DRAG
+        ) {
+            this.setCurrency();
+        }
+
+        const trade_definition_block = this.workspace
+            .getAllBlocks(true)
+            .find(block => block.type === 'trade_definition');
+        if (!trade_definition_block) {
+            return;
+        }
+
+        const market_block = trade_definition_block.getChildByType('trade_definition_market');
+        const trade_type_block = trade_definition_block.getChildByType('trade_definition_tradetype');
```

Comment:
`trade_definition_trade_type` maybe?

```suggestion
        const trade_type_block = trade_definition_block.getChildByType('trade_definition_trade_type');
```

Reply:
Since we're following `trade_type` on line 91 & 92

Reply:
we have trade_definition_CHILDBLOCK types in one word
this is an extract from trade_defintion.js (lines 136-141):
    ```
const market_block = block.getChildByType('trade_definition_market');
const trade_type_block = block.getChildByType('trade_definition_tradetype');
const contract_type_block = block.getChildByType('trade_definition_contracttype');
const candle_interval_block = block.getChildByType('trade_definition_candleinterval');
const restart_on_error_block = block.getChildByType('trade_definition_restartonerror');
const restart_on_buy_sell_block = block.getChildByType('trade_definition_restartbuysell');
```